lsp-pwsh.el

```diff
@@ -321,40 +317,42 @@ Must not nil.")
                 (not (memq major-mode lsp-pwsh--major-modes)))
               '((name . --force-post-completion-for-pwsh))))
 
-;;; Utils
-(defconst lsp-pwsh-unzip-script "\"%s\" -noprofile -noninteractive -nologo -ex bypass -command Expand-Archive -Path '%s' -DestinationPath '%s'"
-  "Powershell script to unzip vscode extension package file.")
-
-(defconst lsp-pwsh-editor-svcs-dl-script "\"%s\" -noprofile -noninteractive -nologo -ex bypass -command Invoke-WebRequest -UseBasicParsing -uri '%s' -outfile '%s'"
-  "Command executed via `shell-command' to download the latest PowerShellEditorServices release.")
-
 (defcustom lsp-pwsh-github-asset-url
   "https://github.com/%s/%s/releases/latest/download/%s"
   "GitHub latest asset template url."
   :type 'string
   :group 'lsp-pwsh
   :package-version '(lsp-mode . "6.2"))
 
-(defun lsp-pwsh--get-extension (url dest)
-  "Get extension from URL and extract to DEST."
-  (let ((temp-file (make-temp-file "ext" nil ".zip")))
-    ;; since we know it's installed, use powershell to download the file
-    ;; (and avoid url.el bugginess or additional libraries)
-    (shell-command (format lsp-pwsh-editor-svcs-dl-script lsp-pwsh-exe url temp-file))
-    (message "lsp-pwsh: Downloading done!")
-    (if (file-exists-p dest) (delete-directory dest 'recursive))
-    (shell-command (format lsp-pwsh-unzip-script lsp-pwsh-exe temp-file dest))
-    (message "lsp-pwsh: Finish unzip!")))
-
-(defun lsp-pwsh-setup (&optional forced)
+(defun lsp-pwsh-setup (_client callback update)
   "Downloading PowerShellEditorServices to `lsp-pwsh-dir'.
 FORCED if specified with prefix argument."
-  (interactive "P")
-  (let ((parent-dir (file-name-directory lsp-pwsh-dir)))
-    (unless (and (not forced) (file-exists-p parent-dir))
-      (lsp-pwsh--get-extension
-       (format lsp-pwsh-github-asset-url "PowerShell" "PowerShellEditorServices" "PowerShellEditorServices.zip")
-       parent-dir))))
+
+  (unless (and lsp-pwsh-exe (file-executable-p lsp-pwsh-exe))
+    (user-error "Use `lsp-pwsh-exe' with the value of `%s' is not a valid powershell binary"
+                lsp-pwsh-exe))
+
+  (let ((parent-dir (file-name-directory lsp-pwsh-dir))
+        (url (format lsp-pwsh-github-asset-url "PowerShell"
+                     "PowerShellEditorServices" "PowerShellEditorServices.zip"))
+        (temp-file (make-temp-file "ext" nil ".zip")))
+    (unless (and (not update) (file-exists-p parent-dir))
```

Comment:
Since you're merging `lsp-pwsh-setup` and `lsp-pwsh--get-extension`, it would be great if you can make some minor changes too.
This should check for `(f-exists? lsp-pwsh-dir)`